packages/compiler/core/library.ts

```diff
@@ -37,14 +38,29 @@ export function createCadlLibrary<
   T extends { [code: string]: DiagnosticMessages },
   E extends Record<string, any>
 >(lib: Readonly<CadlLibraryDef<T, E>>): CadlLibrary<T, E> {
+  let emitterOptionValidator: JSONSchemaValidator;
+
   const { reportDiagnostic, createDiagnostic } = createDiagnosticCreator(lib.diagnostics, lib.name);
   function createStateSymbol(name: string): symbol {
     return Symbol.for(`${lib.name}.${name}`);
   }
 
   const caller = getCaller();
   loadedUrls.add(caller);
-  return { ...lib, reportDiagnostic, createDiagnostic, createStateSymbol };
+  return {
+    ...lib,
+    reportDiagnostic,
+    createDiagnostic,
+    createStateSymbol,
+    get emitterOptionValidator() {
```

Comment:
Is this a perf change?

Reply:
Yes, we would recreate this every time we loaded an emitter and it is very slow.

Reply:
ah interesting